test/utils/create_resources.go

```diff
@@ -67,6 +68,7 @@ func IsRetryableAPIError(err error) bool {
 	return false
 }
 
+// CreatePodWithRetries creates a new Pod retrying on failure
```

Comment:
- `retrying on failure` -> `while retrying on failure.`
- please use consistent case for API objects (Pod, DaemonSet, ...) either have them all MixedCase or lower case.

applies to all retry methods bellow.